src/rqt_robot_steering/robot_steering.py

```diff
@@ -149,8 +149,13 @@ def _on_topic_changed(self, topic):
             self._publisher = rospy.Publisher(topic, Twist)
 
     def _on_stop_pressed(self):
-        self._widget.x_linear_slider.setValue(0)
-        self._widget.z_angular_slider.setValue(0)
+        # If the current value of sliders is zero directly send stop twist msg
+        if self._widget.x_linear_slider.value() is 0 and \
+                self._widget.z_angular_slider.value() is 0:
+            self._publisher.publish(Twist())
+        else:
+            self._widget.x_linear_slider.setValue(0)
+            self._widget.z_angular_slider.setValue(0)
```

Comment:
I think the idea to publish the message immediately when stop is being pressed is a good idea. I would suggest the following though:

* continue to set the slider values to zero (as before)
* invoke `self._on_parameter_changed()` explicitly
* in between also set `self.zero_cmd_sent = False` to ensure allow repeatedly sending zero messages when the stop button is pressed explicitly
```